### PR TITLE
Adding a new partition mode for querier

### LINKED ISSUE #22: Additive Windows should reset when in partition mode

### PR DESCRIPTION
**Pull request by akshaisarma**
Fixes #22 

Adds a new concept when creating a Querier so that we don't need to expose the the isClosedForPartition (and the new resetForPartition).

### CODE REVIEW

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 1]
```diff
@@ -117,7 +120,7 @@
  *     else
  *         if (q.isClosedForPartition())
```
Comment (hbxie) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: isClosedForPartition -> isClosed

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 2]
```diff
@@ -133,7 +136,7 @@
  *         remove q
  *     if (q.isClosedForPartition())
```
Comment (hbxie) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: ditto

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 3]
```diff
@@ -117,7 +120,7 @@
  *     else
  *         if (q.isClosedForPartition())
  *             emit(q.getData())
- *             q.reset()
+ *             q.resetForPartition()
```
Comment (hbxie) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: Still reset, no resetForPartion defined in this class.

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 4]
```diff
@@ -133,7 +136,7 @@
  *         remove q
  *     if (q.isClosedForPartition())
  *         emit(q.getData())
- *         q.reset()
+ *         q.resetForPartition()
```
Comment (hbxie) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: ditto

**Comment by akshaisarma** [comment]
@hbxie Done

File: src/main/java/com/yahoo/bullet/querying/Querier.java [hunk 5]
```diff
@@ -276,6 +293,10 @@
     // This is counting the number of times we get the data out of the query.
     private RateLimiter rateLimit;
 
+    // Mode for the querier
+    private Mode mode;
+
+
```
Comment (NathanSpeidel) [inline src/main/java/com/yahoo/bullet/querying/Querier.java]: Etxra line?

**Pull request merged**